packages/web-components/src/components/tabs-extended/tabs-extended.ts

```diff
@@ -223,6 +249,23 @@ class DDSTabsExtended extends MediaQueryMixin(
     this._isLTR = window.getComputedStyle(this).direction === 'ltr';
     this._activeTabIndex = parseInt(this._activeTab, 10);
 
+    // Reset state and event listeners when switching views
+    if (changedProperties.has('_isLargeOrGreater')) {
+      // Removing event listener if in mobile view, else adding the event listener to any other view that's not mobile.
+      if (!_isLargeOrGreater) {
+        _tabItems.forEach((tab) => {
+          tab.removeEventListener(
+            'click',
+            this._handleAccordionClick.bind(this)
+          );
+        });
+      } else {
+        _tabItems.forEach((tab) => {
+          tab.addEventListener('click', this._handleAccordionClick.bind(this));
+        });
+      }
+    }
```

Comment:
Hrm, can you help me understand why we need this? 

My understanding is that when we cross over the `lg` breakpoint and `this._isLargerOrGreater` gets updated, we end up re-rendering the whole component. As a result, `this._tabItems` gets updated, which causes another update cycle. At that point we have fresh DOM elements, add we only the event listener if we're below the breakpoint.

Reply:
Hey @jkaeser,
I needed to add this piece of code because the accordion items were working the first time they're rendered in mobile, but if you switch the viewport to desktop and then go back to mobile, it stops working. I think the changes in the viewport were not properly attaching the events to the accordion items. When I wrote this code in the updated() method, it fixed that bug by ensuring the proper Event Listeners were added and removed based on the viewport. I took a look at the code several times and could not find out why changing the viewport made the accordion items not to work. Do you see something I'm not seeing? Thanks a lot in advance for all your help!

Reply:
Buckle up, this is a lot 😆 

I tracked the event listeners on a `<dds-tab>` element as I crossed back and forth over the breakpoint. What's actually happening is that any time we cross the threshold, a new copy of the `click` event listener is added. None are ever removed.

All of the accumulated event listeners fire off their callbacks in rapid succession any time an accordion item is clicked. When there is only one listener, we get the expected result. When there is a second, the functionality breaks; the first callback fires and changes `this._activeTabIndex`, and then the second fires and sets it back to the clicked tab's index value. Add a third listener, and things work again. It keeps going like this, with even numbers of listeners working and odd numbers failing.

<img width="371" alt="Screenshot 2024-03-15 at 3 01 58 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/12532727/0bf50225-9947-480f-99d2-0c741d46570b">

The real fix to all of this is making sure only one copy of the event listener is ever attached. In my opinion, this can be best achieved by removing this new chunk of code and modifying the existing logic. Two potential approaches come to mind:

1. A few lines down in the conditional before adding the event listener, you can check for the absence of an attribute (e.g. `dds-processed`) on the `<dds-tab>` elements, then set that attribute after adding the event listener. This way the conditional will only succeed once.
   - ```ts
      if (!tab.hasAttribute('dds-processed') && !_isLargeOrGreater) {
        tab.addEventListener(...);
        tab.setAttribute('dds-processed', true);
      }
      ```
2. You can remove the existing `tab.addEventListener` and instead use an approach called [Event Delegation](https://javascript.info/event-delegation). You'll attach a single click handler to the `<dds-tabs-extended>` element, check if the element is what we're actually looking for, and only then fire off `this._handleAccordionClick`.
   - ```ts
      @HostListener('click')
      protected _handleClick(event) {
        const { target } = event;
        if (target.matches('button.bx--accordion__heading') {
          this._handleAccordionClick(event);
        }
      }
      ```

Your call on whichever looks better! As always, let me know if you want to go over any of this together.